src/Reflex/Spider/Internal.hs

```diff
@@ -1641,9 +1666,12 @@ mergeCheap d = Event $ \sub -> do
           , subscriberInvalidateHeight = \_ -> return ()
           , subscriberRecalculateHeight = \_ -> return ()
           }
-    (EventSubscription _ changeSubd, change) <- subscribeAndRead (dynamicUpdated d) s
+    (EventSubscription sln changeSubscription, change) <- subscribeAndRead (dynamicUpdated d) s
     forM_ change $ \c -> defer $ SomeMergeUpdate (updateMe c) invalidateMyHeight revalidateMyHeight
-    liftIO $ writeIORef changeSubdRef (s, changeSubd)
+    -- We explicitly hold on to the unsubscribe function from subscribing to the update event.
+    -- If we don't do this, there are certain cases where mergeCheap will fail to properly retain
+    -- its subscription.
+    liftIO $ writeIORef changeSubdRef (s, sln, changeSubscription)
```

Comment:
Let's not unpack and re-pack the EventSubscription, in line with the new comment above.